src/pagination.ts

```diff
@@ -141,23 +162,31 @@ export class CursorPaginator<T> {
     });
   }
 
+  private getHasNextAndPrevious(result: { entities: T[]; raw: any[] }, pageSize: number, first: number | undefined, after: string | undefined, before: string | undefined) {
+    const hasMore = result.raw.length > pageSize
+    const showingFirstPage = first !== undefined
+    return showingFirstPage
+        ? { hasNext: hasMore, hasPrevious: !!after }
+        : { hasNext: !!before, hasPrevious: hasMore }
+  }
+
   /**
    * Encode a cursor for the given row.
    */
-  cursor(row: Row) {
+  cursor(row: Row): string {
```

Comment:
I started adding some return types in this file

Reply:
I prefer annotations, too. It makes it easy to see at a glance what to expect!